archaius-core/src/test/java/com/netflix/config/DynamicMapPropertyTest.java

```diff
@@ -0,0 +1,38 @@
+package com.netflix.config;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+public class DynamicMapPropertyTest {
+    
+    public class DynamicIntegerSetProperty extends DynamicSetProperty<Integer> {
+        public DynamicIntegerSetProperty(String propName, Set<Integer> defaultValue) {
+            super(propName, defaultValue);
+        }
+
+        @Override
+        protected Integer from(String value) {
+            return Integer.parseInt(value);
+        }
+    }
+    
+
+    @Test
+    public void testCallbacksAddUnsubscribe() {
```

Comment:
For test cases, can you change the initial set value to an unordered list so we can verify the new map, set property actually preserves the "random" order?